src/boolean_result.cpp

```diff
@@ -809,6 +810,40 @@ Manifold::Impl Boolean3::Result(OpType op) const {
   PRINT(n12 << " new verts from edgesP -> facesQ");
   PRINT(n21 << " new verts from facesP -> edgesQ");
 
+  const int numProp = std::max(inP_.NumProp(), inQ_.NumProp());
+  if (numProp > 0) {
+    // Ideally, switch over from triProperties vec to Halfedge.startProp - then
+    // the prop index will be available in the append edges functions below.
+    Vec<std::atomic<int>> prop2VertP(inP_.NumPropVert());
+    for_each_n(autoPolicy(inP_.NumTri()), countAt(0), inP_.NumTri(),
+               [&](size_t tri) {
+                 for (const int i : {0, 1, 2}) {
+                   prop2VertP[inP_.meshRelation_.triProperties[tri][i]].store(
+                       inP_.halfedge_[3 * tri + i].startVert);
+                 }
```

Comment:
Another piece of 2-i: No - every propVert maps to exactly one vertex, hence this vector being possible. Whenever the same value shows up multiple times in this list, that's a vertex that's associated with multiple propVerts.

Reply:
This is an important feature (because it's also implicitly a restriction in MeshGL), so to answer question 1 - we definitely need to duplicate rather than reuse the index. Again, this duplication is actually exceedingly rare in practice since it's only for handing self-intersecting meshes.